tests/python/rest_api/test_projects.py

```diff
@@ -1389,10 +1411,17 @@
                 project["id"], patched_project_write_request={"assignee_id": new_assignee_id}
             )
 
-            if new_assignee_id == old_assignee_id:
-                assert updated_project.assignee_updated_date == project["assignee_updated_date"]
+            op = operator.eq if new_assignee_id == old_assignee_id else operator.ne
+
+            # FUTURE-TODO: currently it is possible to have a project with assignee but with assignee_updated_date == None
+            # because there were no migration to set some assignee_updated_date for such projects/tasks/jobs
+            if isinstance(updated_project.assignee_updated_date, datetime):
+                assert op(
+                    str(updated_project.assignee_updated_date.isoformat()).replace("+00:00", "Z"),
+                    project["assignee_updated_date"],
+                )
             else:
-                assert updated_project.assignee_updated_date != project["assignee_updated_date"]
+                assert op(updated_project.assignee_updated_date, project["assignee_updated_date"])
```

Comment:
I prefer the current approach, because it is more compact, and there is no need to duplicate logic with handling `isinstance(... datetime)` in both branches.